training/training.module

```diff
@@ -313,6 +380,128 @@ function training_form_user_login_block_alter(&$form, &$form_state, $form_id) {
 }
 
 /**
+ * Implements hook_field_formatter_info().
+ */
+function training_field_formatter_info() {
+  return array(
+    'text_formated' => array(
+      'lable' => t('Formated text'),
+      'field types' => array('text'),
+      'settings' => array(
+        'format' => '%s'
+      ),
+    ),
+  );
+}
+
+/**
+ * Implements hook_field_formatter_settings_form().
+ */
+function training_field_formatter_settings_form($field, $instance, $view_mode, $form, &$form_state) {
+  $display = $instance['display'][$view_mode];
+  $settings = $display['settings'];
+  $element = array();
+  if ($display['type'] == 'text_formatted') {
+    $element['format'] = array(
+      '#title' => t('Format'),
+      '#type' => 'textfield',
+      '#size' => 20,
+      '#default_value' => $settings['format'],
+      '#required' => TRUE,
+    );
+  }
+
+  return $element;
+}
+
+/**
+ * Implements hook_field_formatter_settings_summary().
+ */
+function training_field_formatter_settings_summary($field, $instance, $view_mode) {
+  $display = $instance['display'][$view_mode];
+  $settings = $display['settings'];
+  $summary = t('Use a "@format" format to display this field', array(
+    '@format' => $settings['format'],
+  ));
+
+  return $summary;
+}
+
+/**
+ * Implements hook_field_formatter_view().
+ */
+function training_field_formatter_view($entity_type, $entity, $field, $instance, $langcode, $items, $display) {
+  $element = array();
+  $settings = $display['settings'];
+  switch ($display['type']) {
+    case 'text_formated':
+      foreach ($items as $delta => $item) {
+        $element[$delta] = array('#markup' => sprintf($settings['format'], $item['value']));
+      }
+      break;
+  }
+
+  return $element;
+}
+
+/**
+ * Implements hook_entity_info().
+ */
+function training_entity_info() {
+  $return = array(
+    'training_item' => array(
+      'label' => t('Task'),
+      'entity class' => 'Entity',
+      'controller class' => 'EntityAPIController',
+      'base table' => 'training',
+      'fieldable' => FALSE,
+      'entity keys' => array(
+        'id' => 'id',
+      ),
+      'bundle keys' => array(),
+      'bundles' => array(),
+      'load hook' => 'training_item_load',
+      'view modes' => array(),
+      'label callback' => 'entity class label',
+      'uri callback' => 'entity class uri',
+      'module' => 'training',
+      'access callback' => 'training_access',
+    ),
+    'text_msg' => array(
+      'label' => t('Text message'),
+      'entity class' => 'Entity',
+      'controller class' => 'EntityAPIController',
+      'base table' => 'text_message',
+      'fieldable' => FALSE,
+      'entity keys' => array(
+        'id' => 'id',
+      ),
+      'bundle keys' => array(),
+      'bundles' => array(),
```

Comment:
remove, empty array is a default value, doesn't make sense to declare it.

This entry can be omitted if this entity type exposes a single bundle (all entities have the same collection of fields).
https://api.drupal.org/api/drupal/modules%21system%21system.api.php/function/hook_entity_info/7